### PR TITLE
First use of Nimbus FML plugin

### PR DESCRIPTION
**Pull request by jhugman**
This requires the newly landed Nimbus FML plugin.

Closes #22925 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: nimbus.fml.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      - channel: nightly
+        value: {
+          "sections-enabled": {
+            "top-sites": true,
+            "jump-back-in": true,
+            "recently-saved": true,
+            "recent-explorations": true,
+            "pocket": true
+          }
+        }
```
Comment (Amejia481) [inline nimbus.fml.yaml:33]: Is this going to affect the values returned from nimbus on [experiments.getVariables(FeatureId.HOME_PAGE, false)](https://github.com/mozilla-mobile/fenix/blob/6f9e725ec58f0658bb6cde057782991395bc3485/app/src/main/java/org/mozilla/fenix/experiments/NimbusFeatures.kt#L77)? At the moment we are [enabling everything by default](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/experiments/NimbusFeatures.kt#L52-L57), except for pocket that it's only enabled in `"en-US"` and `"en-CA" ` .

**Review by Amejia481 — COMMENTED** [review]
This looks good in general just one question to clarify

File: nimbus.fml.yaml (hunk 1 above)
Comment (jhugman) [reply nimbus.fml.yaml:33]: > At the moment we are [enabling everything by default](https://github.com/mozilla-mobile/fenix/blob/main/app/src/main/java/org/mozilla/fenix/experiments/NimbusFeatures.kt#L52-L57)

This is exactly how it should work: I'll update the file to reflect the current state.

Comment (Amejia481) [reply nimbus.fml.yaml:33]: 👍🏽 , just to confirm, did the last push update the default values?

**Comment by jhugman** [comment]
This is ready for reviewing and landing; /cc @Amejia481

**Comment by Amejia481** [comment]
> This is ready for reviewing and landing; /cc @Amejia481

Awesome thanks for updating it. I'll take a look later in the day :)

**Comment by Amejia481** [comment]
@jhugman in general it looks good, I was trying to do some manual testing locally but I was unable, I got a build error while trying:

```
Task :app:nimbusFeaturesDebug
Nimbus FML generating Kotlin
manifest   nimbus.fml.yaml
class      org.mozilla.fenix.nimbus.FxNimbus
channel    developer
R.class    org.mozilla.fenix.R
outputFile app/build/generated/source/nimbus/debug/kotlin/org/mozilla/fenix/nimbus/FxNimbus.kt

> Task :app:compileDebugKotlin FAILED
w: ATTENTION!
This build uses unsafe internal compiler arguments:

-XXLanguage:+NonParenthesizedAnnotationsOnFunctionalTypes

This mode is not recommended for production use,
as no stability/compatibility guarantees are given on
compiler or generated code. Use it at your own risk!

e: app/src/main/java/org/mozilla/fenix/ext/Nimbus.kt: (20, 46): No value passed for parameter 'transform'



FAILURE: Build failed with an exception.

```


Additionally, it will be good if we could squash the commits.

**Comment by Amejia481** [comment]
> @jhugman in general it looks good, I was trying to do some manual testing locally but I was unable, I got a build error while trying:
> 
> ```
> Task :app:nimbusFeaturesDebug
> Nimbus FML generating Kotlin
> manifest   nimbus.fml.yaml
> class      org.mozilla.fenix.nimbus.FxNimbus
> channel    developer
> R.class    org.mozilla.fenix.R
> outputFile app/build/generated/source/nimbus/debug/kotlin/org/mozilla/fenix/nimbus/FxNimbus.kt
> 
> > Task :app:compileDebugKotlin FAILED
> w: ATTENTION!
> This build uses unsafe internal compiler arguments:
> 
> -XXLanguage:+NonParenthesizedAnnotationsOnFunctionalTypes
> 
> This mode is not recommended for production use,
> as no stability/compatibility guarantees are given on
> compiler or generated code. Use it at your own risk!
> 
> e: app/src/main/java/org/mozilla/fenix/ext/Nimbus.kt: (20, 46): No value passed for parameter 'transform'
> 
> 
> 
> FAILURE: Build failed with an exception.
> ```
> 
> Additionally, it will be good if we could squash the commits.

Same build failure is on CI too https://firefoxci.taskcluster-artifacts.net/ddVWA51lQYy4KmvVcRYnxw/0/public/logs/live_backing.log

**Comment by eliserichards** [comment]
Removing the mergify label so this doesn't block the queue. This needs to be r+'d before you can add the label 👍

**Review by Amejia481 — APPROVED** [review]
image

Awesome work 👏🏽  !

**Comment by mcomella** [comment]
I believe this PR regressed our COLD MAIN time to first frame benchmark by 100ms: https://github.com/mozilla-mobile/fenix/issues/23492#issuecomment-1026264201

**Pull request merged**